e2e/nomostest/wait_for_sync.go

```diff
@@ -379,36 +368,34 @@ func (nt *NT) WaitForRepoSyncStalledError(rsNamespace, rsName, reason, message s
 		})
 }
 
-// TODO: Uncomment when Syncing condition consistency is fixed
-// func validateRootSyncError(statusErrs []v1beta1.ConfigSyncError, syncingCondition *v1beta1.RootSyncCondition, code string, message string, expectedErrorSourceRefs []v1beta1.ErrorSource) error {
-// 	if err := testpredicates.ValidateError(statusErrs, code, message); err != nil {
-// 		return err
-// 	}
-// 	if syncingCondition == nil {
-// 		return fmt.Errorf("syncingCondition is nil")
-// 	}
-// 	if syncingCondition.Status == metav1.ConditionTrue {
-// 		return fmt.Errorf("status.conditions['Syncing'].status is True, expected false")
-// 	}
-// 	if !reflect.DeepEqual(syncingCondition.ErrorSourceRefs, expectedErrorSourceRefs) {
-// 		return fmt.Errorf("status.conditions['Syncing'].errorSourceRefs is %v, expected %v", syncingCondition.ErrorSourceRefs, expectedErrorSourceRefs)
-// 	}
-// 	return nil
-// }
-
-// TODO: Uncomment when Syncing condition consistency is fixed
-// func validateRepoSyncError(statusErrs []v1beta1.ConfigSyncError, syncingCondition *v1beta1.RepoSyncCondition, code string, message string, expectedErrorSourceRefs []v1beta1.ErrorSource) error {
-// 	if err := testpredicates.ValidateError(statusErrs, code, message); err != nil {
-// 		return err
-// 	}
-// 	if syncingCondition == nil {
-// 		return fmt.Errorf("syncingCondition is nil")
-// 	}
-// 	if syncingCondition.Status == metav1.ConditionTrue {
-// 		return fmt.Errorf("status.conditions['Syncing'].status is True, expected false")
-// 	}
-// 	if !reflect.DeepEqual(syncingCondition.ErrorSourceRefs, expectedErrorSourceRefs) {
-// 		return fmt.Errorf("status.conditions['Syncing'].errorSourceRefs is %v, expected %v", syncingCondition.ErrorSourceRefs, expectedErrorSourceRefs)
-// 	}
-// 	return nil
-// }
+func validateRootSyncError(statusErrs []v1beta1.ConfigSyncError, syncingCondition *v1beta1.RootSyncCondition, code string, message string, expectedErrorSourceRefs []v1beta1.ErrorSource, resources []v1beta1.ResourceRef) error {
+	if err := testpredicates.ValidateError(statusErrs, code, message, resources); err != nil {
+		return err
+	}
+	if syncingCondition == nil {
+		return fmt.Errorf("syncingCondition is nil")
+	}
+	if syncingCondition.Status == metav1.ConditionTrue {
+		return fmt.Errorf("status.conditions['Syncing'].status is True, expected false")
+	}
+	if !reflect.DeepEqual(syncingCondition.ErrorSourceRefs, expectedErrorSourceRefs) {
```

Comment:
Do we want to use semantic DeepEqual here?

Reply:
It's an array of strings, so I'm not sure whether it matters. Do you have a preference?

Reply:
In general it's good to avoid using reflect in prod code. In test code it's less important, but usually slightly faster. The only real behavioral difference here would be accepting nil as equal to an empty list, but that's probably not important here.

Reply:
done